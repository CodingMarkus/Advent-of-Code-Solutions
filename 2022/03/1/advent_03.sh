#!/bin/sh

# Run as:
# cat advent_03_sample.txt | sh advent_03.sh
# cat advent_03_input.txt  | sh advent_03.sh

set -e

getItems()
{
	while read -r line
	do
		lineLen=${#line}
		halfLineLen=$(( lineLen / 2 ))
		comp1=$( printf '%s' "$line" | head -c $halfLineLen )
		comp2=${line#"$comp1"}
		printf "%s" "$comp2" | tr -C -d "$comp1"
		printf "\n"
	done
}


ord()
{
  LC_CTYPE=C printf '%d' "'$1"
}


sum=0
for item in $( getItems )
do
	ordVal=$( ord "$item" )
	if [ "$ordVal" -ge 97 ] && [ "$ordVal" -le 122 ]
	then
		sum=$(( sum + ordVal - 96 ))
	elif [ "$ordVal" -ge 65 ] && [ "$ordVal" -le 90 ]
	then
		sum=$(( sum + ordVal - 64 + 26 ))
	else
		echo "Error"
		exit 1
	fi
done
echo "Sum: $sum"