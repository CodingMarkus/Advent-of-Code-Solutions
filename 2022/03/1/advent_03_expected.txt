Sum: 157
Sum: 7568
